iml-agent/src/action_plugins/stratagem/action_purge.rs

```diff
@@ -5,31 +5,31 @@
 use crate::{agent_error::ImlAgentError, http_comms::mailbox_client};
 use futures::future::poll_fn;
 use futures::{Future, Stream};
+use liblustreapi::LlapiFid;
 use tokio_threadpool::blocking;
 
 pub fn purge_files(
     device: &str,
     args: impl IntoIterator<Item = String>,
 ) -> Result<(), ImlAgentError> {
-    let mntpt = match liblustreapi::search_rootpath(&device) {
-        Ok(m) => m,
-        Err(e) => return Err(ImlAgentError::LiblustreError(e)),
-    };
-    liblustreapi::rmfids(&mntpt, args).map_err(ImlAgentError::LiblustreError)
+    let llapi = LlapiFid::create(&device).map_err(|e| {
+        log::error!("Failed to find rootpath({}) -> {}", device, e);
+        ImlAgentError::LiblustreError(e)
+    })?;
+    llapi.rmfids(args).map_err(ImlAgentError::LiblustreError)
 }
 
-fn search_rootpath(device: String) -> impl Future<Item = String, Error = ImlAgentError> {
+fn search_rootpath(device: String) -> impl Future<Item = LlapiFid, Error = ImlAgentError> {
     poll_fn(move || {
-        blocking(|| liblustreapi::search_rootpath(&device))
-            .map_err(|_| panic!("the threadpool shut down"))
+        blocking(|| LlapiFid::create(&device)).map_err(|_| panic!("the threadpool shut down"))
     })
     .and_then(std::convert::identity)
     .from_err()
 }
 
-fn rm_fids(mntpt: String, fids: Vec<String>) -> impl Future<Item = (), Error = ImlAgentError> {
+fn rm_fids(llapi: LlapiFid, fids: Vec<String>) -> impl Future<Item = (), Error = ImlAgentError> {
     poll_fn(move || {
-        blocking(|| liblustreapi::rmfids(&mntpt, fids.clone()))
+        blocking(|| llapi.clone().rmfids(fids.clone()))
```

Comment:
Nit: using `Arc::clone(&llapi)` makes it more clear that this just creating a new reference to the pointer and is thus not expensive.